app/src/main/java/com/keylesspalace/tusky/fragment/NotificationPreferencesFragment.kt

```diff
@@ -0,0 +1,107 @@
+ *
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>. */
+
+package com.keylesspalace.tusky.fragment
+
+import android.os.Bundle
+import android.support.v14.preference.SwitchPreference
+import android.support.v7.preference.Preference
+import android.support.v7.preference.PreferenceFragmentCompat
+import android.view.View
+import com.keylesspalace.tusky.R
+import com.keylesspalace.tusky.db.AccountManager
+import com.keylesspalace.tusky.di.Injectable
+import javax.inject.Inject
+
+
+class NotificationPreferencesFragment : PreferenceFragmentCompat(), Preference.OnPreferenceChangeListener, Injectable {
+
+    @Inject
+    lateinit var accountManager: AccountManager
+
+    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
+        addPreferencesFromResource(R.xml.notification_preferences)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val activeAccount = accountManager.activeAccount
+
+        if (activeAccount != null) {
+
+            val notificationPref = findPreference("notificationsEnabled") as SwitchPreference
+            notificationPref.isChecked = activeAccount.notificationsEnabled
+            notificationPref.onPreferenceChangeListener = this
+
+            val mentionedPref = findPreference("notificationFilterMentions") as SwitchPreference
+            mentionedPref.isChecked = activeAccount.notificationsMentioned
+            mentionedPref.onPreferenceChangeListener = this
+
+            val followedPref = findPreference("notificationFilterFollows") as SwitchPreference
+            followedPref.isChecked = activeAccount.notificationsFollowed
+            followedPref.onPreferenceChangeListener = this
+
+            val boostedPref = findPreference("notificationFilterReblogs") as SwitchPreference
+            boostedPref.isChecked = activeAccount.notificationsReblogged
+            boostedPref.onPreferenceChangeListener = this
+
+            val favoritedPref = findPreference("notificationFilterFavourites") as SwitchPreference
+            favoritedPref.isChecked = activeAccount.notificationsFavorited
+            favoritedPref.onPreferenceChangeListener = this
+
+            val soundPref = findPreference("notificationAlertSound") as SwitchPreference
+            soundPref.isChecked = activeAccount.notificationSound
+            soundPref.onPreferenceChangeListener = this
+
+            val vibrationPref = findPreference("notificationAlertVibrate") as SwitchPreference
+            vibrationPref.isChecked = activeAccount.notificationVibration
+            vibrationPref.onPreferenceChangeListener = this
+
+            val lightPref = findPreference("notificationAlertLight") as SwitchPreference
+            lightPref.isChecked = activeAccount.notificationLight
+            lightPref.onPreferenceChangeListener = this
```

Comment:
That's a lot of boilerplate 🙂

Reply:
We can invent a function or use something like `apply` but it doesn't worth it I guess